### PR TITLE
Some basic models for archive service

### PR DESCRIPTION
**Pull request by yma955**
I extract some from my work, this is just the basic model entries if you want to enhance some other features based on it.

### CODE REVIEW

File: src/main/java/org/commonjava/indy/service/archive/GreetingResource.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package org.commonjava.indy.service;
+package org.commonjava.indy.service.archive;
 
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/archive/GreetingResource.java:4]: I think this is just setup resource from quarkus, so it should not be needed anymore.

File: src/main/java/org/commonjava/indy/service/archive/config/ConfigResource.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package org.commonjava.indy.service.config;
+package org.commonjava.indy.service.archive.config;
 
 import org.eclipse.microprofile.config.inject.ConfigProperty;
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/archive/config/ConfigResource.java:4]: I think this is just setup resource from quarkus, so it should not be needed anymore.

File: src/test/java/org/commonjava/indy/service/archive/GreetingResourceTest.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package org.commonjava.indy.service;
```
Comment (ligangty) [inline src/test/java/org/commonjava/indy/service/archive/GreetingResourceTest.java:1]: I think this class is not needed anymore.

File: src/test/java/org/commonjava/indy/service/archive/NativeGreetingResourceIT.java [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package org.commonjava.indy.service;
+package org.commonjava.indy.service.archive;
```
Comment (ligangty) [inline src/test/java/org/commonjava/indy/service/archive/NativeGreetingResourceIT.java:2]: I think this class is not needed anymore.

File: src/main/java/org/commonjava/indy/service/archive/model/dto/HistoricalContentDTO.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+ */
+package org.commonjava.indy.service.archive.model.dto;
+
+import java.util.Set;
+
+public class HistoricalContentDTO
+{
+
+    private Set<HistoricalContentEntryDTO> downloads;
+
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/archive/model/dto/HistoricalContentDTO.java:27]: Not sure if uploads are not needed here or just missed.

**Review by ligangty — COMMENTED** [review]
Basically looks good. Some comments see reviews.

File: src/main/java/org/commonjava/indy/service/archive/model/dto/HistoricalContentDTO.java (hunk 5 above)
Comment (yma955) [reply src/main/java/org/commonjava/indy/service/archive/model/dto/HistoricalContentDTO.java:27]: This is used to do obj mapper for the list PNC provided, if sidecar will redirect all maven uploads to remote Indy, I see uploads are not needed in archive generation.

File: src/main/java/org/commonjava/indy/service/archive/GreetingResource.java (hunk 1 above)
Comment (yma955) [reply src/main/java/org/commonjava/indy/service/archive/GreetingResource.java:4]: These classes were created by @shokakucarrier , feel free to remove in the following commit if no further usage for you, here are just package refactor.

Comment (shokakucarrier) [reply src/main/java/org/commonjava/indy/service/archive/GreetingResource.java:4]: I think it should be removed, it's generated by quarkus, I've made the change on my commit, so you can remove this change on code and wait for mine

**Comment by shokakucarrier** [comment]
Are we sure to not use the same model in the indy source code? Gang mentioned it is here 

https://github.com/Commonjava/indy/blob/master/addons/folo/model-java/src/main/java/org/commonjava/indy/folo/model/TrackedContent.java

**Comment by yma955** [comment]
@shokakucarrier HistoricalContentDTO will be one wrapping tracking.

**Pull request merged**